src/object.jl

```diff
@@ -144,6 +144,46 @@ function _download_task(io; kw...)
     return nothing
 end
 
+_ndownload_tasks(total_size, part_size, numthreads=Threads.nthreads()) =
+    min(numthreads, max(1, div(total_size, part_size, RoundUp)))
+
+"""
+    PrefetchedDownloadStream{T <: Object} <: IO
```

Comment:
I think people usually (or just often?) put constructors here instead of / in addition to what you have.

I think that would be useful, especially since there's different ones.